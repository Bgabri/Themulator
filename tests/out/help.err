==1385007== Memcheck, a memory error detector
==1385007== Copyright (C) 2002-2024, and GNU GPL'd, by Julian Seward et al.
==1385007== Using Valgrind-3.23.0 and LibVEX; rerun with -h for copyright info
==1385007== Command: ./thml.out --help
==1385007== 
==1385007== 
==1385007== HEAP SUMMARY:
==1385007==     in use at exit: 0 bytes in 0 blocks
==1385007==   total heap usage: 2 allocs, 2 frees, 8,192 bytes allocated
==1385007== 
==1385007== All heap blocks were freed -- no leaks are possible
==1385007== 
==1385007== For lists of detected and suppressed errors, rerun with: -s
==1385007== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
