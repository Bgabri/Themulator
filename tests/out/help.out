Usage: thml [options] <command>
Commands:
	download
		Starts an interface to login and download files from Themis.
	run
		Compiles and runs the program with the given option.
	interpret
		interpret the program with the given option.
Options:
	--help, -h
		Display this information
	--compiler, -c
		The compiler or interpreter to use
	--compiler-flags, -f
		The flags to pass to the compiler
	--dry-run, -d
		run without command execution
	--verbose
		print out all the execution steps
	--valgrind, -v
		runs the binary through valgrind
	--bin-name <name>, -b <name>
		the name to give to the outputted binary
	--dir <folder>, -D <folder>
		the folder of the source files to judge
	--in-dir, -I <folder>
		the folder to the input of the test cases
	--out-dir <folder>, -O <folder>
		the folder where the output is placed
		will also be used as the cookie output path
	--bin-dir <folder>, -B <folder>
		the folder to output the binaries to
	--ref-dir <folder>, -R <folder>
		the folder with the reference outputs of the test cases
